components/resourceDetails/resourceSharing/advancedSharingButton/index.jsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* eslint-disable react/forbid-prop-types */
+
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { createStyles, Modal } from "@material-ui/core";
+import { useBem } from "@solid/lit-prism-patterns";
+import clsx from "clsx";
+import { makeStyles } from "@material-ui/styles";
+import { Button } from "@inrupt/prism-react-components";
+import AgentPickerModal from "../agentPickerModal";
+import styles from "./styles";
+import usePolicyPermissions from "../../../../src/hooks/usePolicyPermissions";
+import usePermissionsWithProfiles from "../../../../src/hooks/usePermissionsWithProfiles";
+
+const useStyles = makeStyles((theme) => createStyles(styles(theme)));
+
+export const TESTCAFE_ID_ADVANCED_SHARING_BUTTON = "advanced-sharing-button";
+const TESTCAFE_ID_MODAL_OVERLAY = "modal-overlay";
```

Comment:
Minor nitpick: I like to include the name of the component in the `TESTCAFE_ID`s to avoid naming collisions, e.g. I would change `TESTCAFE_ID_MODAL_OVERLAY = "modal-overlay"` to `TESTCAFE_ID_ADVANCED_SHARING_BUTTON_MODAL_OVERLAY = "advanced-sharing-button-modal-overlay"`. It's a bit verbose, and not strictly necesssary, so I'll leave it to you to decide if you want to do it.